File: src/1207_unique-number-of-occurences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        num_occurences = Counter(arr)
        occurences_set = set()

        for num in set(arr):
            count = num_occurences[num]

            if count in occurences_set:
                return False
            else:
                occurences_set.add(count)

        return True

File: Elixir/1207_unique-number-of-occurences.ex
defmodule Solution do
  @spec unique_occurrences(arr :: [integer]) :: boolean
  def unique_occurrences(arr) do
    frequencies = Enum.frequencies(arr) |> Enum.map(fn {_, freq} -> freq end)
    size(MapSet.new(frequencies)) == length(frequencies)
  end
end
